src/MockBot.UnitTests/Controllers/ChatControllerTests.cs

```diff
@@ -83,44 +80,62 @@ public void ShouldCreateAndReturnChat()
         [InlineData("Some text")]
         public async Task ShouldAddMessageToChatAsync(string payload)
         {
+            // Arrange
             _sut.ControllerContext = new ControllerContext()
             {
                 HttpContext = GetContext(payload)
             };
 
-            var message = new ChatMessage(payload);
-            var chat = new Chat();
+            var chat = _chatService.CreateChat();
             var currentDateTime = new DateTime();
-            _ = _mockChatService.Setup(mock => mock.AddMessage(chat.Id, payload)).Returns(message);
 
+            // Act
             var result = await _sut.PostMessageAsync(chat.Id).ConfigureAwait(false);
 
+            // Assert
             var createdResult = Assert.IsType<CreatedResult>(result);
             var resultMessage = Assert.IsType<ChatMessage>(createdResult.Value);
             Assert.Equal($"/chats/{chat.Id}/messages", createdResult.Location);
             Assert.NotEmpty(resultMessage.Id.ToString());
             Assert.True(currentDateTime < resultMessage.CreatedAt);
-            _mockChatService.Verify(mock => mock.AddDmrRequest(message));
-            _mockDmrService.Verify(mock => mock.Enqueue(It.IsAny<DmrRequest>()));
         }
 
         [Theory]
         [InlineData("Some text")]
         public async Task ShouldFailToAddMessageToChatWhenNoChatWithGivenIdAsync(string payload)
         {
+            // Arrange
             _sut.ControllerContext = new ControllerContext()
             {
                 HttpContext = GetContext(payload)
             };
-            var chat = new Chat();
-            _ = _mockChatService.Setup(mock => mock.AddMessage(chat.Id, payload)).Throws(new ArgumentOutOfRangeException(chat.Id.ToString()));
+            var chatId = Guid.NewGuid();
 
+            // Act
+            var result = await _sut.PostMessageAsync(chatId).ConfigureAwait(false);
 
-            var result = await _sut.PostMessageAsync(chat.Id).ConfigureAwait(false);
-
+            // Assert
             _ = Assert.IsType<NotFoundObjectResult>(result);
         }
 
+        [Fact]
+        public async Task ShouldReturnBadReqestIfBodyEmpty()
+        {
+            // Arrange
+            _sut.ControllerContext = new ControllerContext()
+            {
+                HttpContext = GetContext(string.Empty)
+            };
+            var chatId = Guid.NewGuid();
+
+            // Act
+            BadRequestObjectResult result = (BadRequestObjectResult)await _sut.PostMessageAsync(chatId).ConfigureAwait(false);
+
+            // Assert
+            _ = Assert.IsType<BadRequestObjectResult>(result);
```

Comment:
I might be clueless here, but isn't `result` quite definitely of type BadRequestObjectResult from everything done on line 132?

Reply:
Yes, it is, but there is no harm in keeping that assertion although it will never be false.

The reason we have to cast to `BadRequestObjectResult` is to be able to access the `.Values` property and assert that the message is what was expected.

I'll look to see if there is a cleaner way to do this.

Reply:
I've changed it so that the result is cast after it has been asserted as the right type. The value of `Assert.IsType<BadRequestObjectResult>(result);` is still questionable but probably OK to keep in here.